onadata/libs/permissions.py

```diff
@@ -357,10 +363,36 @@ def get_team_project_default_permissions(team, project):
 
 
 def filter_queryset_xform_meta_perms(xform, user, instance_queryset):
-
     if user.has_perm(CAN_VIEW_XFORM_ALL, xform) or xform.shared_data:
         return instance_queryset
     elif user.has_perm(CAN_VIEW_XFORM_DATA, xform):
         return instance_queryset.filter(user=user)
     else:
         return instance_queryset.none()
+
+
+def filter_queryset_xform_meta_perms_sql(xform, user, query):
+    if user.has_perm(CAN_VIEW_XFORM_ALL, xform) or xform.shared_data:
+        ret_query = query
+    elif user.has_perm(CAN_VIEW_XFORM_DATA, xform):
+        try:
+            if query and isinstance(query, six.string_types):
+                query = json.loads(query)
+                if isinstance(query, list):
+                    query = query[0]
+
+            else:
+                query = dict()
+                query.update({"_submitted_by": user.username})
+                ret_query = json.dumps(query)
+
+        except (ValueError, AttributeError):
+            query_list = list()
+            query_list.append({"_submitted_by": user.username})
+            query_list.append(query)
+
+            ret_query = json.dumps(query_list)
+    else:
+        ret_query = NoRecordsPermission()
```

Comment:
I think Exceptions are meant to be raised you should probably have a 
```python

raise NoRecordsPermission()
```

You can then catch the exception in `xform_viewset.py`  as:
```python
try:
    query = filter_queryset_xform_meta_perms_sql(self.get_object(), self.request.user, query)
except NoRecordsPermission:
    pass
```